sql/analyzer/resolve_ctes.go

```diff
@@ -24,7 +24,7 @@ import (
 	"github.com/dolthub/go-mysql-server/sql/transform"
 )
 
-const maxCteDepth = 5
+const maxCteDepth = 10
```

Comment:
`depth` takes on a new meaning here. Afaik if someone using the previous version of the code had one CTE top level reference once in a 10th level nested subquery, they will experience a new error? If we increment depth only when recursing in the CTE case would it preserve the previous behavior?